mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -145,13 +135,20 @@ class LogisticRegressionModel (
        */
       var bestClass = 0
       var maxMargin = 0.0
-      var i = 0
-      while(i < margins.size) {
-        if (margins(i) > maxMargin) {
-          maxMargin = margins(i)
+      val withBias = if (dataMatrix.size + 1 == dataWithBiasSize) true else false
```

Comment:
In Multinomial Logistic Regression, the intercepts will be parts of the weights.
So we need to get whether the model add intercepts and will be checked only once at here rather than in each iteration..